hypothesis_sqlalchemy/columnar/records.py

```diff
@@ -46,8 +46,11 @@ def lists_factory(columns: List[Column],
                       if is_column_unique(column)]
 
     if unique_indices:
-        def unique_by(row: Tuple[Any, ...]) -> Tuple[Any, ...]:
-            return tuple(row[index] for index in unique_indices)
+        # Create a tuple of functions, each function asserting the uniqueness
+        # of a single column value
+        unique_by = tuple(
+            (lambda row, idx=idx: row[idx]) for idx in unique_indices
+        )
```

Comment:
can this be replaced using [`itemgetter`](https://docs.python.org/3/library/operator.html#operator.itemgetter)? something like
```python
from operator import itemgetter
...
unique_by = tuple(map(itemgetter, unique_indices))
```
?

Reply:
Personally, I feel `itemgetter` is often less readable, but I will see if it's easy to implement like that